test/parallel/test-fs-cp.mjs

```diff
@@ -943,10 +946,13 @@ if (!isWindows) {
 
 // It rejects if options.mode is invalid.
 {
+  const src = './test/fixtures/copy/kitchen-sink';
+  const dest = nextdir();
   await assert.rejects(
-    fs.promises.cp('a', 'b', {
+    fs.promises.cp(src, dest, mustNotMutateObjectDeep({
+      recursive: true,
```

Comment:
Why recursive is needed now?

Reply:
Without it, the test throws `ERR_FS_EISDIR` instead of `ERR_OUT_OF_RANGE`.

Reply:
So this is a breaking change, right?

Reply:
Thinking about it now, I think so because if someone was checking for an `mode` error in `fs.cp` synchronously, this behavior would change to async. We could keep using `getValidMode` on the JS side just for `fs.cp` which is the only problematic function or don't make these changes at all. Or maybe there is another way, idk.